### PR TITLE
collect + analyze

### PR DESCRIPTION
**Pull request by calebjclark**
A PR that's waaaaaaaaaaaaaaaay tooooooooooo biiiiiiiiiiiiig...

- Major restructuring of tests/checks from a single flow that all the checks use into a collect + analyze architecture that uses plugins within each to isolate and test distinct flows.

- `user-agents` npm package been replaced with our own real-user-agents package included in DA. Fake user-agent strings are no longer included since we use real user-agents imported from our Slab data.

- profiler package has already been removed in favor of profiles imported from Slab.

### CODE REVIEW

File: analyze/lib/Plugin.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        x.layerId === layer.id && (!x.useragentIds.length || x.useragentIds.includes(useragentId))
+      );
+    });
+
+    for (const probeBucket of probeBuckets) {
+      for (const probe of probeBucket.probes) {
+        if (probe.checkType !== checkType) continue;
+        const toCheck = checksById[probe.check.id];
+        const humanScore = probe.check.generateHumanScore(toCheck, profileCountOverTime);
+        // if (probe.id === 'func-aenw') {
```
Comment (blakebyrnes) [inline analyze/lib/Plugin.ts:68]: cleanup?

File: analyze/plugins/tls-clienthello/ToDo.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+The following extensions are not checked:
```
Comment (blakebyrnes) [inline analyze/plugins/tls-clienthello/ToDo.txt:1]: Does this mean the content is not checked? Or that the presence/order is not checked?

Comment (calebjclark) [reply analyze/plugins/tls-clienthello/ToDo.txt:1]: Shoot... I have no idea....

File: analyze/lib/Plugin.ts (hunk 1 above)
Comment (calebjclark) [reply analyze/lib/Plugin.ts:68]: I've been using this quite a bit during development... unless you have a problem, I'm going to leave it for now.

File: analyze/plugins/tls-clienthello/ToDo.txt (hunk 2 above)
Comment (calebjclark) [reply analyze/plugins/tls-clienthello/ToDo.txt]: Removed.

**Pull request merged**